source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -126,6 +128,14 @@ Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMa
   } else {
     config_->stats().not_compressed_.inc();
   }
+
+  // Even if we decided not to compress due to incompatible Accept-Encoding value,
+  // the Vary header would need to be inserted to let a caching proxy in front of Envoy
+  // know that the requested resource still can be served with compression applied.
```

Comment:
I think we should put this in `docs/root/configuration/http/http_filters/compressor_filter.rst` as well. WDYT?

Reply:
Right! I overlooked the doc needs to be corrected too. Will update.

Reply:
@dio Updated!